src/lib/polling.js

```diff
@@ -0,0 +1,53 @@
+import {Observable} from "rxjs/Observable";
+import 'rxjs/add/operator/repeatWhen';
+import 'rxjs/add/operator/retryWhen';
+import 'rxjs/add/operator/scan';
+import 'rxjs/add/operator/delay';
+
+let retryStrategy = ({attempts, delay}) => {
+	return (errors) => {
+		return errors.scan((accumulator, value) => {
+			accumulator += 1;
+			if (accumulator < attempts) {
+				return accumulator;
+			}
+			throw new Error(value);
+		}, 0).delay(delay);
+	};
+};
```

Comment:
Does this traverse through the errors to figure out if there is a problem? Not 100% sure what this does...